hist/histv7/inc/ROOT/RAxis.hxx

```diff
@@ -258,69 +254,66 @@ public:
          return 2;
    };
 
-   /// Get the bin index for the underflow bin (or the last bin outside range
+   /// Get the bin index for the underflow bin (or `kInvalidBin`
    /// if CanGrow()).
    int GetUnderflowBin() const noexcept {
       if (CanGrow())
-         return -1;
+         return kInvalidBin;
       else
-         return 0;
+         return kUnderflowBin;
    }
 
-   /// Get the bin index for the underflow bin (or the next bin outside range
+   /// Get the bin index for the overflow bin (or `kInvalidBin`
    /// if CanGrow()).
    int GetOverflowBin() const noexcept {
       if (CanGrow())
-         return GetNBins();
+         return kInvalidBin;
       else
-         return GetNBins() - 1;
+         return kOverflowBin;
    }
 
-   /// Whether the bin index is referencing a bin lower than the axis range.
-   bool IsUnderflowBin(int bin) const noexcept { return bin <= GetUnderflowBin(); }
+   /// Get the bin index for the first bin of the axis
+   int GetFirstBin() const noexcept { return 1; }
 
-   /// Whether the bin index is referencing a bin higher than the axis range.
-   bool IsOverflowBin(int bin) const noexcept { return bin >= GetOverflowBin(); }
+   /// Get the bin index for the last bin of the axis
+   int GetLastBin() const noexcept { return GetNBinsNoOver(); }
 
    ///\name Iterator interfaces
    ///\{
 
-   /// Get a const_iterator pointing to the first non-underflow bin.
-   const_iterator begin() const noexcept { return const_iterator{GetUnderflowBin() + 1}; }
-
-   /// Get a const_iterator pointing the first bin, whether underflow or not
-   const_iterator begin_with_underflow() const noexcept { return const_iterator{0}; }
+   /// Get a const_iterator pointing to the first regular bin.
+   const_iterator begin() const noexcept { return const_iterator{GetFirstBin()}; }
 
-   /// Get a const_iterator pointing right beyond the last non-overflow bin
-   /// (i.e. pointing to the overflow bin, if any).
-   const_iterator end() const noexcept { return const_iterator{GetOverflowBin()}; }
-
-   /// Get a const_iterator pointing right beyond the last bin, whether overflow or not
-   const_iterator end_with_overflow() const noexcept { return const_iterator{GetNBins()}; }
+   /// Get a const_iterator pointing to the last regular bin
```

Comment:
Just making sure this works: `end()` must point *beyond* the last bin, else `for (auto iter = axis.begin(); iter != axis.end(); ++iter)` will not include the last bin.

Please make sure that the code corresponds to that (do we have a test?) and adjust the comment:
```suggestion
   /// Get a const_iterator pointing beyond the last regular bin
```